### ISSUE TITLE
[Bug]: Screen reader not announcing position of IconButton despite defining aria-posinset and aria-setsize

### ISSUE THREAD

**Issue by hgangavalli** [microsoft/fluentui#28513]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
-
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/button#fluent-ui-v9.  Give aria-posinset and aria-setsize values to iconButton props and verify screen reader behaviour

## Actual Behavior
After defining posinset and setsize, screen reader is not announcing position of button
## Expected Behavior
Screen reader should announce "x of y" depending on position of button.

Tried with following code and observed issue persists
 ```
<IconButton iconProps={emojiIcon} title="Emoji" ariaLabel="Emoji" disabled={disabled} checked={checked} 
aria-posinset={1} aria-setsize={2}
/>       
  <IconButton           
      menuProps={menuProps}           
      iconProps={emojiIcon}           
      title="Emoji"           
     ariaLabel="Emoji"           
     disabled={disabled}           
     checked={checked}           
     aria-posinset={2}           
     aria-setsize={2}         
/>
```
Observed that in given demo https://developer.microsoft.com/en-us/fluentui#/controls/web/button#fluent-ui-v9, screen reader announces position for split button as 1 of 1. 

### Requested priority

Normal

### Products/sites affected

eDiscovery

### Are you willing to submit a PR to fix?

no
### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by spmonahan** [comment]
I believe this is incorrect usage of `aria-posinset` and `aria-setsize`. From the MDN docs on `aria-posinset`:

> The aria-posinset attribute defines an element's number or position in the current set of listitems or treeitems when not all items are present in the DOM.

Similarly for `aria-setsize`:

> The aria-setsize attribute defines the number of items in the current set of listitems or treeitems when not all items in the set are present in the DOM.

These attributes are only valid on listitems and treeitems so cannot be used for a button as in this example.
